### PR TITLE
Machine ID: Add Protos for Bot Resource and CRUD RPCs

### PR DESCRIPTION
**Pull request by strideynet**
Part of https://github.com/gravitational/teleport/issues/33808

Based on guidance in the draft Resource Guidelines RFD: https://github.com/gravitational/teleport/pull/34103

Introduces the protos for a Bot resource and a BotService that will provide CRUD for the Bot resource.

I've named the package `machineidv1` and the service `BotService` fairly intentionally - I'd like to introduce some other functionality at a future data (e.g tbot heartbeating / command and control) and I think it would be nice to have these defined within the same Proto package but implemented as other services. This lets us clearly mark which RPCs are expected be called by users and which should be called by tbot.

### CODE REVIEW

File: api/proto/teleport/machineid/v1/bot_service.proto [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+// The request for CreateBot.
+message CreateBotRequest {
+  // The bot to create.
+  Bot bot = 1;
+}
+
+// The response for CreateBot.
+message CreateBotResponse {
```
Comment (strideynet) [inline api/proto/teleport/machineid/v1/bot_service.proto:59]: There's some debate over on the RFD about whether these "wrapper" messages should be used or if the resource should just be the RPCs direct return. We may wish to wait for a decision to be made before we merge this PR.

File: api/proto/teleport/machineid/v1/bot_service.proto [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  rpc ListBots(ListBotsRequest) returns (ListBotsResponse);
+  // CreateBot is used to create a Bot.
+  //
+  // This will return an error if a Bot by that name already exists.
+  rpc CreateBot(CreateBotRequest) returns (CreateBotResponse);
+  // UpdateBot is used to modify an existing Bot.
+  rpc UpdateBot(UpdateBotRequest) returns (UpdateBotResponse);
+  // UpsertBot is used to create or replace an existing Bot.
+  //
+  // Prefer using CreateBot and UpsertBot.
```
Comment (nklaassen) [inline api/proto/teleport/machineid/v1/bot_service.proto:44]:
```suggestion
  // Prefer using CreateBot and UpdateBot.
```

**Review by nklaassen — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]
This all seems very reasonable. I don't have much feedback to offer on the wrapper message discussion though.

File: api/proto/teleport/machineid/v1/bot.proto [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  // Where multiple specified with the same name, these will be merged by the
+  // server.
+  repeated Trait traits = 2;
+}
+
+// Fields that are set by the server as results of operations. These should not
+// be modified by users.
+message BotStatus {
+  // The name of the user associated with the bot.
+  string user = 1;
```
[the remark this answers is not in the document]
Comment (strideynet) [reply api/proto/teleport/machineid/v1/bot.proto:67]: I like that.

File: api/proto/teleport/machineid/v1/bot_service.proto (hunk 1 above)
Comment (strideynet) [reply api/proto/teleport/machineid/v1/bot_service.proto:59]: Great - I'll drop the wrappers from here.

File: api/proto/teleport/machineid/v1/bot.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (strideynet) [reply api/proto/teleport/machineid/v1/bot.proto]: [fcff785](https://github.com/gravitational/teleport/pull/35216/commits/fcff78555031f4365dd9bfbe053a91c6bfea1d91)

**Review by ryanclark — APPROVED** [review]
bot

File: api/proto/teleport/machineid/v1/bot_service.proto (hunk 1 above)
Comment (strideynet) [reply api/proto/teleport/machineid/v1/bot_service.proto]: [fcff785](https://github.com/gravitational/teleport/pull/35216/commits/fcff78555031f4365dd9bfbe053a91c6bfea1d91)

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**